_includes/embed/audio.html

```diff
@@ -0,0 +1,43 @@
+{% assign src = include.src | strip %}
+{% assign title = include.title | strip %}
+{% assign types = include.types | default: '' | strip | split: '|' %}
+
+{% unless src contains '://' %}
+  {%- capture src -%}
+    {% include img-url.html src=src %}
+  {%- endcapture -%}
+{% endunless %}
+
+<p>
+  <audio class="embed-audio" controls>
+    {% assign parts = src | split: '/' %}
+    {% assign filename = parts | last %}
+    {% assign extension = filename | split: '.' | last %}
+    {% assign basename = filename | split: '.' | pop | join: '.' %}
+    {% assign path = parts | pop | join: '/' %}
+
+    {% assign media_item = site.data.media | find: 'extension', extension %}
+    {% if media_item %}
+      <source src="{{ src }}" type="audio/{{ media_item['mimeType'] }}">
+    {% else %}
+      <source src="{{ src }}" type="audio/{{ extension }}">
+    {% endif %}
+
+    {% for extra_type in types %}
+      {% assign filename = basename | append: '.' | append: extra_type %}
+      {% assign extra_src = path | append: '/' | append: filename %}
+      {% assign media_item = site.data.media | find: 'extension', extra_type %}
+      {% if media_item %}
+        <source src="{{ extra_src }}" type="audio/{{ media_item['mimeType'] }}">
+      {% else %}
+        <source src="{{ extra_src }}" type="audio/{{ extra_type }}">
+      {% endif %}
+    {% endfor %}
```

Comment:
There is a lot of duplicated code in `audio.html` and `video.html`, and it may be necessary to add a common module to be referenced by both files (via `{% include %}`).

Reply:
> Thanks @MrMurdock11, I understand your motivation and am in favor of this new feature.
> 
> But this PR contains some unnecessary files. It's not a good practice to overuse `{% include %}`, it will slow down the Jekyll build.

After this comment, I removed all the `{% include.... %}` and rebuilt the template for working without tag `include`

Reply:
Even if we don't use `{% include %}` here, the following redundant code can be avoided

```liquid 

    {% assign media_item = site.data.media | find: 'extension', extension %}
    {% if media_item %}
      <source src="{{ src }}" type="audio/{{ media_item['mimeType'] }}">
    {% else %}
      <source src="{{ src }}" type="audio/{{ extension }}">
    {% endif %}

    {% for extra_type in types %}
      {% assign filename = basename | append: '.' | append: extra_type %}
      {% assign extra_src = path | append: '/' | append: filename %}
      {% assign media_item = site.data.media | find: 'extension', extra_type %}
      {% if media_item %}
        <source src="{{ extra_src }}" type="audio/{{ media_item['mimeType'] }}">
      {% else %}
        <source src="{{ extra_src }}" type="audio/{{ extra_type }}">
      {% endif %}
    {% endfor %}
```

Reply:
as you wish @cotes2020. I'll move it to sources template again.

Reply:
Thanks for your patience @MrMurdock11, whether or not you end up moving to `sources`, the snippet I mentioned above should still be improved.

Reply:
The way I figured it out is to add the `type` from the `src` to the `types`, then generate the HTML in a forloop.

Reply:
Thank you for the feedback, @cotes2020! Ok, I get your point
```liquid
<video class="embed-video file" {{ poster }} {{ attributes }}>
    {% assign parts = video_url | split: '/' %}
    {% assign filename = parts | last %}
    {% assign extension = filename | split: '.' | last %}
    {% assign basename = filename | split: '.' | pop | join: '.' %}
    {% assign path = parts | pop | join: '/' %}

    {% assign types = types | push: extension %}

    {% for type in types %}
      {% assign filename = basename | append: '.' | append: type %}
      {% assign extra_src = path | append: '/' | append: filename %}
      {% assign media_item = site.data.media | find: 'extension', type %}
      {% if media_item %}
        <source src="{{ extra_src }}" type="video/{{ media_item['mime_type'] }}">
      {% else %}
        <source src="{{ extra_src }}" type="video/{{ type }}">
      {% endif %}
    {% endfor %}

    Your browser does not support the video tag. Here is a
    <a href="{{ video_url | strip }}">link to the video file</a> instead.
  </video>
```